crates/sui-indexer-alt-framework/src/pipeline/concurrent/mod.rs

```diff
@@ -233,6 +243,20 @@ pub(crate) fn pipeline<H: Handler + Send + Sync + 'static>(
     let pruner_cancel = cancel.child_token();
     let handler = Arc::new(handler);
 
+    let main_reader_lo = Arc::new(SetOnce::<AtomicU64>::new());
+    // The collector does not need to be lower-bounded by the main reader lo task if it is not part
+    // of a tasked indexer. Set to 0 so the collector does not have to wait for initialization.
+    if task.is_none() {
+        main_reader_lo.set(AtomicU64::new(0)).ok();
+    }
```

Comment:
This is leaky -- you already have logic inside the `track_reader_lo` task that will perform this operation if `task.is_none()`, you don't need this here.